skimage/segmentation/slic_superpixels.py

```diff
@@ -245,6 +245,13 @@ def slic(image, n_segments=100, compactness=10., max_num_iter=10, sigma=0,
     float_dtype = utils._supported_float_type(image.dtype)
     image = image.astype(float_dtype, copy=False)
 
+    # Rescale image to [0, 1] to make choice of compactness insensitive to
+    # input image scale.
+    image -= image.min()
```

Comment:
Since no copy is made during the conversion to float (in the lines above) this will modify the input image :confused:

Reply:
Good catch. Fortunately, I did this same subtraction correctly in the recent `denoise_bilateral` PR :)

Reply:
I changed the astype call above to just use copy=True